### PR TITLE
[stdlib] Port StdlibUnittest to FreeBSD.

### PR DESCRIPTION
**Pull request by dcci**
We almost got there :)

### CODE REVIEW

**Comment by lattner** [comment]
Very cool!  @gribozavr is probably the best one to look at this, but he is out of town until next week.  Maybe @dabrahams could also look?

File: stdlib/private/StdlibUnittest/StdlibUnittest.swift.gyb [hunk 1]
```diff
@@ -16,7 +16,7 @@ import SwiftPrivateDarwinExtras
 
 #if os(OSX) || os(iOS) || os(watchOS) || os(tvOS)
 import Darwin
-#elseif os(Linux)
+#elseif os(Linux) || os(FreeBSD)
 import Glibc
```
Comment (gribozavr) [inline stdlib/private/StdlibUnittest/StdlibUnittest.swift.gyb]: I'm OK with this in the short term, but as we discussed before, Glibc is not the right module name on FreeBSD.

**Comment by gribozavr** [comment]
@dcci Thank you!  For a full port, you need to modify `OSVersion`, `_getOSVersion()` and `TestRunPredicate`, and add a test like `validation-test/stdlib/StdlibUnittestLinux.swift`.

**Comment by dcci** [comment]
Uh, OK, thanks Dmitri and Chris, I'll update the pull request accordingly. BTW, is there an easy way to run a single test in the swift test suite? In LLVM I'd run llvm-lit -sv $testpath but it doesn't seem to do the trick here (unless I'm missing something).

**Comment by jrose-apple** [comment]
Take a look at Testing.rst. It's not *quite* that simple but it's close. IIRC there's also a --filter option for bare lit.

(We don't have an llvm-lit because we have multiple testing configs. I guess we could do one that defaults to the host target.)

**Comment by dcci** [comment]
Update pull request after comments.

File: validation-test/stdlib/StdlibUnittestFreeBSD.swift [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+import ObjectiveC
+#endif
+
+_setOverrideOSVersion(.FreeBSD)
+_setTestSuiteFailedCallback() { print("abort()") }
+
+var XFailsFreeBSD = TestSuite("XFailsFreeBSD")
+
+// CHECK: [   UXPASS ] XFailsFreeBSD.xfail iOS passes{{$}}
+XFailsFreeBSD.test("xfail iOS passes").xfail(.FreeBSDAny(reason: "")).code {
```
Comment (gribozavr) [inline validation-test/stdlib/StdlibUnittestFreeBSD.swift]: `s/iOS/FreeBSD/` ?

**Comment by dcci** [comment]
Updated fail messages -> s/iOS/FreeBSD.

File: stdlib/private/StdlibUnittest/StdlibUnittest.swift.gyb [hunk 3]
```diff
@@ -1180,6 +1187,10 @@ public enum TestRunPredicate : CustomStringConvertible {
     case LinuxAny(reason: let reason):
       return "LinuxAny(*, reason: \(reason))"
     }
```
Comment (gribozavr) [inline stdlib/private/StdlibUnittest/StdlibUnittest.swift.gyb]: This brace terminates the switch statement, it shouldn't be present.  Does the code build for you?  It fails for me.

Comment (dcci) [inline stdlib/private/StdlibUnittest/StdlibUnittest.swift.gyb]: hmm, it does for me, weirdly. Apparently something went wrong when I rebased and updated the pull request. Let me try again.

**Comment by dcci** [comment]
Updated. Hopefully no changes got lost this time. I also tried to squash the two commits in one. Thanks!

**Comment by gribozavr** [comment]
@dcci Thanks, tests pass now!
